### PR TITLE
gh-115119: Detect _decimal dependencies using pkg-config

### PR DESCRIPTION
**Pull request by erlend-aasland**
Only libmpdec 4.0.0 supports pkg-config.

* Issue: gh-115119

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Tested with the following scenarios:

```console
$ ./configure && make                 # Uses CPython vendored mpdecimal
$ ./configure --with-system-libmpdec  # pkg-config detects Homebrew installed libmpdec 4.0.0
$ LIBMPDEC_CFLAGS="-I$HOME/install" LIBMPDEC_LIBS="-L$HOME/install -lm -lmpdec" \
  ./configure --with-system-libmpdec  # Build with custom libmpdec 2.5.1
```

For some reason the latter fails, but I suspect it is an issue with my setup.

File: Modules/_decimal/_decimal.c [hunk 1]
```diff
@@ -33,7 +33,15 @@
 #include "pycore_long.h"          // _PyLong_IsZero()
 #include "pycore_pystate.h"       // _PyThreadState_GET()
 #include "complexobject.h"
-#include "mpdecimal.h"
+
+#include <mpdecimal.h>
+
+// Reuse config from mpdecimal.h if present.
+#if defined(MPD_CONFIG_64)
```
Comment (erlend-aasland) [inline Modules/_decimal/_decimal.c:40]:
```suggestion
#if defined(MPD_CONFIG_64) && !defined(CONFIG_64)
```

File: Modules/_decimal/_decimal.c [hunk 2]
```diff
@@ -33,7 +33,15 @@
...
+  #define CONFIG_64 MPD_CONFIG_64
```
Comment (erlend-aasland) [inline Modules/_decimal/_decimal.c]:
```suggestion
  #ifndef CONFIG_64
    #define CONFIG_64 MPD_CONFIG_64
  #endif
```

File: Modules/_decimal/_decimal.c [hunk 3]
```diff
@@ -33,7 +33,15 @@
...
+#elif defined(MPD_CONFIG_32)
+  #define CONFIG_32 MPD_CONFIG_32
```
Comment (erlend-aasland) [inline Modules/_decimal/_decimal.c]:
```suggestion
  #ifndef CONFIG_32
    #define CONFIG_32 MPD_CONFIG_32
  #endif
```

**Review by ned-deily — APPROVED** [review]
LGTM, thanks! FWIW, I was able to successfully build and test with mpdecimal 2.5.1 as well as 4.0.0.

**Comment by erlend-aasland** [comment]
Thanks, Ned! I had forgotten about this :)

**Pull request merged**